File: Data Structures & Algorithms/house-robber-ii/submission-0.py
class Solution:
    def check_rob(self, nums: List[int]) -> int:
        a, b = 0, 0
        for i in range(0, len(nums)):                        
            tmp = b
            b = max(a + nums[i], b)
            a = tmp
        return b
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        return max(self.check_rob(nums[0:-1]), self.check_rob(nums[1::]))

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> umap;
        for (auto i : nums) {
            if (umap[i]) return true;
            umap[i]++;
        }
        return false;
    }
};
